tools/src/main/scala/scala/scalanative/codegen/ResourceEmbedder.scala

```diff
@@ -0,0 +1,199 @@
+    val classpath = config.classPath
+
+    implicit val position: Position = Position.NoPosition
+
+    val foundFiles =
+      if (config.compilerConfig.embedResources) {
+        classpath.flatMap { classpath =>
+          val virtualDir = VirtualDirectory.real(classpath)
+
+          virtualDir.files
+            .flatMap { path =>
+              val pathName =
+                if (!path.toString().startsWith("/")) { // local file
+                  "/" + path.toString()
+                } else { // other file (f.e in jar)
+                  path.toString()
+                }
+
+              if (isInIgnoredDirectory(path)) {
+                config.logger.debug(
+                  s"Did not embed: $pathName. Source file extension detected."
+                )
+                None
+              } else if (isSourceFile(path)) {
+                config.logger.debug(
+                  s"Did not embed: $pathName. File in the ignored scala-native folder."
+                )
+                None
+              } else if (Files.isDirectory(path)) {
+                None
+              } else {
+                Some(ClasspathFile(path, pathName, virtualDir))
+              }
+            }
+        }
+      } else {
+        Seq()
+      }
+
+    @tailrec
+    def filterEqualPathNames(
+        path: List[ClasspathFile],
+        visitedPathNames: Set[String],
+        filteredFiles: List[ClasspathFile]
+    ): List[ClasspathFile] = {
+      path match {
+        case (file @ ClasspathFile(_, pathName, _)) :: tail =>
+          if (visitedPathNames.contains(pathName)) {
+            filterEqualPathNames(tail, visitedPathNames, filteredFiles)
+          } else {
+            filterEqualPathNames(
+              tail,
+              visitedPathNames + pathName,
+              filteredFiles.::(file)
+            )
+          }
+        case Nil =>
+          filteredFiles
+      }
+    }
+
+    val embeddedFiles = filterEqualPathNames(foundFiles.toList, Set.empty, Nil)
+
+    val pathValues = embeddedFiles.map {
+      case ClasspathFile(accessPath, pathName, virtDir) =>
+        val encodedPath = Base64.getEncoder
+          .encode(pathName.toString.getBytes())
+          .map(a => Val.Int(a))
+        Val.ArrayValue(Type.Int, encodedPath.toIndexedSeq)
+    }
+
+    val contentValues = embeddedFiles.map {
+      case ClasspathFile(accessPath, pathName, virtDir) =>
+        val fileBuffer = virtDir.read(accessPath)
+        val encodedContent = Base64.getEncoder
+          .encode(fileBuffer.array())
+          .map(a => Val.Int(a))
+        Val.ArrayValue(Type.Int, encodedContent.toIndexedSeq)
+    }
+
+    def generateArrayVar(name: String, arrayValue: Val.ArrayValue) = {
+      Defn.Var(
+        Attrs.None,
+        extern(name),
+        Type.Ptr,
+        Val.Const(
+          arrayValue
+        )
+      )
+    }
+
+    def generateExtern2DArray(name: String, content: IndexedSeq[Val.Const]) = {
```

Comment:
ArrayValue does not need indexedSeq, regular `Seq` is enough. Changing that would save us from the need to convert all sequences to indexed ones